packages/frontend/core/src/components/app-tabs/app-tabs.tsx

```diff
@@ -0,0 +1,244 @@
+import { IconButton, Loading, observeResize } from '@affine/component';
+import { useAsyncCallback } from '@affine/core/hooks/affine-async-hooks';
+import { DesktopStateSynchronizer } from '@affine/core/modules/workbench/services/desktop-state-synchronizer';
+import type { WorkbenchMeta } from '@affine/electron-api';
+import { apis, events } from '@affine/electron-api';
+import {
+  CloseIcon,
+  DeleteIcon,
+  EdgelessIcon,
+  FolderIcon,
+  PageIcon,
+  PlusIcon,
+  RightSidebarIcon,
+  TagIcon,
+  TodayIcon,
+  ViewLayersIcon,
+} from '@blocksuite/icons/rc';
+import { useServiceOptional } from '@toeverything/infra';
+import { debounce, partition } from 'lodash-es';
+import {
+  Fragment,
+  type MouseEventHandler,
+  type ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import { WindowsAppControls } from '../pure/header/windows-app-controls';
+import * as styles from './app-tabs.css';
+
+type ModuleName = NonNullable<WorkbenchMeta['views'][0]['moduleName']>;
+
+const moduleNameToIcon = {
+  all: <FolderIcon />,
```

Comment:
can this map be reused?

Reply:
Maybe, but I guess right now it is only used in the tabs alone. We can move it out when necessary later.